added general docu about hybrid smart graphs (#742)
arangodb/docs: 3.9/graphs-smart-graphs.md

However the overall performance is never worse than the same query using a
General Graph.

Benefits of Hybrid SmartGraphs
-------------------------------

Hybrid SmartGraphs are capable of using Satellite Collections within their graph
definition. Therefore, defined EdgeDefinitions between SmartCollections and
SatelliteCollections can be created. As Satellite Collections are globally
replicated to each participating DB-Server, graph traversals, shortest path,
weighted traversals and k-shortest-paths queries can be executed partially

> locally on each DB-Server. This means that query execution is guaranteed to be

Reviewer: ```suggestion
locally on each DB-Server. This means a larger part of the query can be executed
```

We should not "guarantee" this here. There may be some code pieces that will not stay local.
e.g. we may introduce an upper limit of Operations per SmartGraph Request to improve LIMIT calls.

fully local whenever data from the Satellite part is required.

> 

Reviewer 2: A note that edge collections between smart- and satellite-collections are replicated like satellite collections (if I got that correctly) seems appropriate, as the amount of space needed could be unexpected otherwise.

can be executed locally on a DB-Server, achieving improved performance for
these type of queries.

Benefits of Hybrid Disjoint SmartGraphs
-------------------------------

Hybrid Disjoint SmartGraphs benefit of all advantages which are present in
Hybrid SmartGraphs. The rules of Disjoint SmartGraphs, described in the chapter
above, stay true.

> 

Reviewer 2: If we now add a comment how edge collections in a hybrid smart graph are replicated, we should also note here that in this case they are replicated like the disjoint smart collections (if I got that correctly), rather than like satellite collections.
